### PR TITLE
Payment method

### PR DESCRIPTION
**Pull request by Barterio**
What: 

Why:

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: satellite/console/consoleweb/consoleql/mutation.go [hunk 1]
```diff
@@ -420,6 +424,95 @@ func rootMutation(log *zap.Logger, service *console.Service, mailService *mailse
...
+				Type: graphql.Boolean,
+				Args: graphql.FieldConfigArgument{
+					FieldProjectID: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+					FieldID:        &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+				},
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					fieldProjectID, _ := p.Args[FieldProjectID].(string)
+					fieldProjectPaymentID, _ := p.Args[FieldID].(string)
+
+					_, err := console.GetAuth(p.Context)
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleql/mutation.go]: Why checking auth here?

File: satellite/console/consoleweb/consoleql/mutation.go [hunk 2]
```diff
@@ -420,6 +424,95 @@ func rootMutation(log *zap.Logger, service *console.Service, mailService *mailse
...
+					FieldProjectID: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+					FieldCardToken: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.String)},
+					FieldIsDefault: &graphql.ArgumentConfig{Type: graphql.NewNonNull(graphql.Boolean)},
+				},
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					projectID, _ := p.Args[FieldProjectID].(string)
+					cardToken, _ := p.Args[FieldCardToken].(string)
+					isDefault, _ := p.Args[FieldIsDefault].(bool)
+
+					auth, err := console.GetAuth(p.Context)
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleql/mutation.go]: Why checking auth here?

File: satellite/console/consoleweb/consoleql/project.go [hunk 3]
```diff
@@ -189,6 +201,35 @@ func graphqlProject(service *console.Service, types *TypeCreator) *graphql.Objec
...
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					project, _ := p.Source.(*console.Project)
+
+					paymentMethods, err := service.GetProjectPaymentMethods(p.Context, project.ID)
+					if err != nil {
+						return nil, err
+					}
+
+					var projectPaymentMethods []projectPayment
+					for _, paymentMethod := range paymentMethods {
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleql/project.go]: Why created new type?

File: satellite/console/projectpayments.go [hunk 4]
```diff
@@ -13,16 +13,33 @@ import (
...
 	PayerID   uuid.UUID
 
 	PaymentMethodID []byte
+	Card            Card
+	IsDefault       bool
 
 	CreatedAt time.Time
 }
+
+type Card struct {
```
Comment (rikysya) [inline satellite/console/projectpayments.go]: Why not to use payments.Card?

File: satellite/console/service.go [hunk 5]
```diff
@@ -165,6 +165,133 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	return u, nil
 }
 
+func (s *Service) AddNewPaymentMethod(ctx context.Context, paymentMethodToken string, isDefault bool, projectID uuid.UUID, userID uuid.UUID) (payment ProjectPayment, err error) {
+	defer mon.Task()(&ctx)(&err)
```
Comment (rikysya) [inline satellite/console/service.go]: Where is the auth check?

File: satellite/console/service.go [hunk 6]
```diff
@@ -165,6 +165,133 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+		Token:      paymentMethodToken,
+		CustomerID: string(info.CustomerID),
+	}
+
+	method, err := s.pm.AddPaymentMethod(ctx, params)
+	if err != nil {
+		return ProjectPayment{}, err
+	}
+
+	projectPaymentInfo := ProjectPayment{ProjectID: projectID,
```
Comment (rikysya) [inline satellite/console/service.go]: Weird formatting

File: satellite/console/service.go [hunk 7]
```diff
@@ -165,6 +165,133 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+		PaymentMethodID: []byte(method.ID),
+		CreatedAt:       time.Now(),
+		IsDefault:       isDefault}
+
+	create, err := s.store.ProjectPayments().Create(ctx, projectPaymentInfo)
+	if err != nil {
+		return ProjectPayment{}, err
+	}
+
+	return *create, nil
```
Comment (rikysya) [inline satellite/console/service.go]: We can return `pointer for consistency`

File: satellite/console/consoleweb/consoleql/project.go [hunk 8]
```diff
@@ -307,6 +348,61 @@ func graphqlProjectUsage() *graphql.Object {
...
+				Type: graphql.DateTime,
+			},
+			FieldIsDefault: &graphql.Field{
+				Type: graphql.Boolean,
+			},
+		},
+	})
+}
+
+type projectPayment struct {
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleql/project.go]: unnecessary struct

File: satellite/console/consoleweb/consoleql/project.go [hunk 9]
```diff
@@ -307,6 +348,61 @@ func graphqlProjectUsage() *graphql.Object {
...
+	// FieldExpirationMonth is field name for expiration month
+	FieldExpirationMonth = "expMonth"
+	// FieldHolderName is field name for holder name
+	FieldHolderName = "holderName"
+	// FieldAddedAt is field name for added at date
+	FieldAddedAt = "addedAt"
+)
+
+// graphqlPaymentMethod creates invoice payment method graphql type
+func graphqlPaymentMethod() *graphql.Object {
```
Comment (rikysya) [inline satellite/console/consoleweb/consoleql/project.go]: This can mimic payments.PaymentMethod

File: satellite/console/service.go [hunk 10]
```diff
@@ -165,6 +165,133 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+
+	projectPayment, err = s.store.ProjectPayments().GetByID(ctx, projectPaymentID)
+	if err != nil {
+		return err
+	}
+	projectPayment.IsDefault = true
+
+	err = s.store.ProjectPayments().Update(ctx, *projectPayment)
+
+	return err
```
Comment (rikysya) [inline satellite/console/service.go]: `return s.store.ProjectPayments().Update(ctx, *projectPayment)`?

File: satellite/console/service.go [hunk 11]
```diff
@@ -165,6 +165,133 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	return s.store.ProjectPayments().Delete(ctx, projectPayment)
+}
+
+// GetProjectPaymentMethods retrieves project payment methods
+func (s *Service) GetProjectPaymentMethods(ctx context.Context, projectID uuid.UUID) ([]ProjectPayment, error) {
```
Comment (rikysya) [inline satellite/console/service.go]: we should return `payments` type

`project_payments` supposed to be just a mapping and implementation detail

File: satellite/console/service.go [hunk 12]
```diff
@@ -627,6 +754,11 @@ func (s *Service) DeleteProjectMembers(ctx context.Context, projectID uuid.UUID,
 		return ErrUnauthorized.Wrap(err)
 	}
 
+	//projectPaymentInfo, err := s.store.ProjectPaymentInfos().GetDefaultByProjctID(ctx, projectID)
```
Comment (rikysya) [inline satellite/console/service.go]: Remove comments

File: satellite/console/service.go [hunk 13]
```diff
@@ -916,7 +1048,8 @@ func (s *Service) CreateMonthlyProjectInvoices(ctx context.Context, date time.Ti
 			continue
 		}
 
-		paymentInfo, err := s.store.ProjectPayments().GetByProjectID(ctx, proj.ID)
+		//paymentInfo, err := s.store.ProjectPayments().GetByProjectID(ctx, proj.ID)
```
Comment (rikysya) [inline satellite/console/service.go]: remove comments

File: satellite/payments/localpayments/local.go [hunk 14]
```diff
@@ -6,6 +6,7 @@ package localpayments
 import (
 	"context"
 	"crypto/rand"
+	"github.com/stripe/stripe-go"
```
Comment (rikysya) [inline satellite/payments/localpayments/local.go]: goimports

File: satellite/payments/localpayments/local.go [hunk 15]
```diff
@@ -59,6 +60,10 @@ type service struct {
 	db DB
 }
 
+func (*service) AddPaymentMethod(ctx context.Context, params payments.AddPaymentMethodParams) (*stripe.PaymentMethod, error) {
```
Comment (rikysya) [inline satellite/payments/localpayments/local.go]: `satellite` should be able to operate without external dependencies

File: satellite/payments/service.go [hunk 16]
```diff
@@ -5,12 +5,14 @@ package payments
 
 import (
 	"context"
+	"github.com/stripe/stripe-go"
```
Comment (rikysya) [inline satellite/payments/service.go]: `goimports`

File: satellite/payments/service.go [hunk 17]
```diff
@@ -5,12 +5,14 @@ package payments
...
 	"time"
 )
 
 // Service is interfaces that defines behavior for working with payments
 type Service interface {
 	CreateCustomer(ctx context.Context, params CreateCustomerParams) (*Customer, error)
+	AddPaymentMethod(ctx context.Context, params AddPaymentMethodParams) (*stripe.PaymentMethod, error)
```
Comment (rikysya) [inline satellite/payments/service.go]: `payments.Service` supposed to be independent from payments provided

File: satellite/payments/stripepayments/stripe.go [hunk 18]
```diff
@@ -90,6 +90,26 @@ func (s *service) GetCustomer(ctx context.Context, id []byte) (_ *payments.Custo
 	}, nil
 }
 
+// AddPaymentMethod add payment method to defined customer
+func (s *service) AddPaymentMethod(ctx context.Context, params payments.AddPaymentMethodParams) (*stripe.PaymentMethod, error) {
+	cparams := &stripe.PaymentMethodParams{
+		Type: stripe.String(string(stripe.PaymentMethodTypeCard)),
+		Card: &stripe.PaymentMethodCardParams{Token: &params.Token},
+		//Customer: &params.CustomerID,
```
Comment (rikysya) [inline satellite/payments/stripepayments/stripe.go]: remove comments

File: satellite/satellitedb/projectpaymentinfos.go [hunk 19]
```diff
@@ -6,54 +6,96 @@ package satellitedb
 import (
 	"context"
 
+	"github.com/zeebo/errs"
```
Comment (rikysya) [inline satellite/satellitedb/projectpaymentinfos.go]: goimports

File: satellite/satellitedb/projectpaymentinfos.go [hunk 20]
```diff
@@ -6,54 +6,96 @@ package satellitedb
...
 
 	"storj.io/storj/satellite/console"
 	dbx "storj.io/storj/satellite/satellitedb/dbx"
 )
 
-// projectpayments is the an implementation of console.ProjectPayments.
+// projectPayments is the an implementation of console.ProjectPayments.
 // Allows to work with project payment info storage
-type projectpayments struct {
+type projectPayments struct {
```
Comment (rikysya) [inline satellite/satellitedb/projectpaymentinfos.go]: why renamed?

File: web/satellite/index.html [hunk 21]
```diff
@@ -8,6 +8,7 @@
   </head>
   <body style="margin: 0px !important; height: 100vh; zoom: 100%">
     <div id="app"></div>
+    <script src="https://js.stripe.com/v3/"></script>
```
Comment (rikysya) [inline web/satellite/index.html]: add dependencies the same way as `analytics` if possible

File: web/satellite/src/api/paymentMethods.ts [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+                            isDefault
+                        }
+                    }
+                }`
+            ),
+            fetchPolicy: 'no-cache',
+            errorPolicy: 'all'
+        }
+    );
+        console.log('fetch pm methods')
```
Comment (rikysya) [inline web/satellite/src/api/paymentMethods.ts]: remove `console.logs`

File: web/satellite/src/components/project/paymentMethods/NewPaymentMethodPopup.vue [hunk 23]
```diff
@@ -0,0 +1,228 @@
...
+
+    @Component(
+        {
+            data: function () {
+                return {
+                    makeDefault: false,
+                };
+            },
+            mounted: function () {
+                if (!window['Stripe']) {
```
Comment (rikysya) [inline web/satellite/src/components/project/paymentMethods/NewPaymentMethodPopup.vue]: `satellite` should be able to operate without external services

File: web/satellite/src/components/project/paymentMethods/NewPaymentMethodPopup.vue [hunk 24]
```diff
@@ -0,0 +1,228 @@
...
+                    this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Stripe library not loaded');
+
+                    return;
+                }
+
+                const stripe = window['Stripe']('pk_test_bXKJTU49iu1dy9Al0iEqlfVc00Ze0m5lXT');
```
Comment (rikysya) [inline web/satellite/src/components/project/paymentMethods/NewPaymentMethodPopup.vue]: API keys should not be exposed

File: satellite/console/service.go (hunk 5 above)
Comment (Barterio) [reply satellite/console/service.go]: At the mutation level

File: satellite/console/service.go (hunk 10 above)
Comment (Barterio) [reply satellite/console/service.go]: yes, good point.

File: satellite/console/consoleweb/consoleql/mutation.go (hunk 1 above)
Comment (Barterio) [reply satellite/console/consoleweb/consoleql/mutation.go]: To prevent it from going to deeper layer to console/service

File: satellite/console/consoleweb/consoleql/mutation.go (hunk 2 above)
Comment (Barterio) [reply satellite/console/consoleweb/consoleql/mutation.go]: same as above

File: satellite/console/projectpayments.go (hunk 4 above)
Comment (Barterio) [reply satellite/console/projectpayments.go]: to reduce dependency tree

File: satellite/console/consoleweb/consoleql/project.go [hunk 25]
```diff
@@ -51,6 +55,14 @@ const (
 	FieldCurrentPage = "currentPage"
 	// FieldTotalCount is a field name for bucket usage count total
 	FieldTotalCount = "totalCount"
+	// FieldCardBrand is a field name for credit card bard
```
Comment (stefanbenten) [inline satellite/console/consoleweb/consoleql/project.go]: brand

File: satellite/console/consoleweb/consoleql/project.go [hunk 26]
```diff
@@ -51,6 +55,14 @@ const (
...
+	FieldCardBrand = "brand"
+	// FieldCardLastFour is a field name for credit card last four digits
+	FieldCardLastFour = "lastFour"
+	// FieldCardToken is a field name for credit card token
+	FieldCardToken = "cardToken"
```
Comment (stefanbenten) [inline satellite/console/consoleweb/consoleql/project.go]: Isnt that called CVC?

File: satellite/console/consoleweb/consoleql/project.go [hunk 27]
```diff
@@ -189,6 +201,35 @@ func graphqlProject(service *console.Service, types *TypeCreator) *graphql.Objec
 					return service.GetBucketTotals(p.Context, project.ID, cursor, before)
 				},
 			},
+			FieldPaymentMethods: &graphql.Field{
+				Type: graphql.NewList(types.paymentMethod),
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					project, _ := p.Source.(*console.Project)
+
+					paymentMethods, err := service.GetProjectPaymentMethods(p.Context, project.ID)
```
Comment (stefanbenten) [inline satellite/console/consoleweb/consoleql/project.go]: I think PaymentOptions is more appropriate here, rather than Methods (which in my definition would be Card, bar, etc.)

Comment (stefanbenten) [reply satellite/console/consoleweb/consoleql/project.go]: Options in a sense of choosing one of the CC cards.

File: satellite/console/service.go [hunk 28]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/zeebo/errs"
 	"go.uber.org/zap"
 	"golang.org/x/crypto/bcrypt"
-	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+	"gopkg.in/spacemonkeygo/monkit.v2"
```
Comment (stefanbenten) [inline satellite/console/service.go]: That change is unnecessary

File: satellite/console/service.go [hunk 29]
```diff
@@ -168,6 +168,141 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	return u, nil
 }
 
+// AddNewPaymentMethod - is a method to add new payment method for project
+func (s *Service) AddNewPaymentMethod(ctx context.Context, paymentMethodToken string, isDefault bool, projectID uuid.UUID) (payment *ProjectPayment, err error) {
```
Comment (stefanbenten) [inline satellite/console/service.go]: AddNewPaymentOption?

File: satellite/console/service.go [hunk 30]
```diff
@@ -168,6 +168,141 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+	create, err := s.store.ProjectPayments().Create(ctx, projectPaymentInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return create, nil
+}
+
+// SetDefaultPaymentMethod - is a method to set default payment method for current project
+func (s *Service) SetDefaultPaymentMethod(ctx context.Context, projectPaymentID uuid.UUID, projectID uuid.UUID) (err error) {
```
Comment (stefanbenten) [inline satellite/console/service.go]: Same as above

File: satellite/payments/localpayments/local.go [hunk 31]
```diff
@@ -9,7 +9,7 @@ import (
 	"time"
 
 	"github.com/zeebo/errs"
-	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+	"gopkg.in/spacemonkeygo/monkit.v2"
```
Comment (stefanbenten) [inline satellite/payments/localpayments/local.go]: Unnecessary Change

File: satellite/payments/stripepayments/stripe.go [hunk 32]
```diff
@@ -8,11 +8,11 @@ import (
 	"fmt"
 	"time"
 
-	stripe "github.com/stripe/stripe-go"
+	"github.com/stripe/stripe-go"
 	"github.com/stripe/stripe-go/client"
 	"github.com/zeebo/errs"
 	"go.uber.org/zap"
-	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+	"gopkg.in/spacemonkeygo/monkit.v2"
```
Comment (stefanbenten) [inline satellite/payments/stripepayments/stripe.go]: Unnecessary change

File: satellite/payments/stripepayments/stripe.go [hunk 33]
```diff
@@ -90,6 +90,40 @@ func (s *service) GetCustomer(ctx context.Context, id []byte) (_ *payments.Custo
 	}, nil
 }
 
+// AddPaymentMethod add payment method to defined customer
+func (s *service) AddPaymentMethod(ctx context.Context, params payments.AddPaymentMethodParams) (*payments.PaymentMethod, error) {
```
Comment (stefanbenten) [inline satellite/payments/stripepayments/stripe.go]: AddPaymentOption

File: satellite/satellitedb/dbx/satellitedb.dbx [hunk 34]
```diff
@@ -273,24 +273,36 @@ read all (
     orderby asc project.name
 )
 
-// add is default, make composite key
 model project_payment (
-    key project_id
```
Comment (stefanbenten) [inline satellite/satellitedb/dbx/satellitedb.dbx]: Do we have a migration for this?

File: satellite/satellitedb/projectpaymentinfos.go (hunk 20 above)
Comment (stefanbenten) [reply satellite/satellitedb/projectpaymentinfos.go]: proper camelcase 👍

File: web/satellite/src/components/project/ProjectPaymentMethods.vue [hunk 35]
```diff
@@ -1,189 +1,61 @@
-// Copyright (C) 2019 Storj Labs, Inc.
+// Copyright (C) 2018 Storj Labs, Inc.
```
Comment (stefanbenten) [inline web/satellite/src/components/project/ProjectPaymentMethods.vue]: Dont change the copyright back??

File: web/satellite/src/utils/apolloManager.ts [hunk 36]
```diff
@@ -9,7 +9,7 @@ import { getToken } from '@/utils/tokenManager';
 
 // Satellite url
 const satelliteUrl = new HttpLink({
-    uri: '/api/graphql/v0',
+    uri: 'http://localhost:10002/api/graphql/v0',
```
Comment (stefanbenten) [inline web/satellite/src/utils/apolloManager.ts]: This should not be hardcoded, right?

File: satellite/console/consoleweb/consoleql/project.go (hunk 26 above)
Comment (Barterio) [reply satellite/console/consoleweb/consoleql/project.go]: Which one?
Card Token? `CardToken` is used with stripe-elements to add payment method.
We use their API and form to add payment method. All validation and other checks are on their side, and in success case we receive card token which later will be process on our backend

File: web/satellite/src/utils/apolloManager.ts (hunk 36 above)
Comment (Barterio) [reply web/satellite/src/utils/apolloManager.ts]: Yes, sure. This is for test purposes only and will be reverted before Merging PR.

File: satellite/console/consoleweb/consoleql/project.go (hunk 26 above)
Comment (Barterio) [reply satellite/console/consoleweb/consoleql/project.go]: > Isnt that called CVC?

We don't store sensitive user data on our side

Comment (stefanbenten) [reply satellite/console/consoleweb/consoleql/project.go]: Okay cool, reminded me on that one

File: satellite/console/service.go [hunk 37]
```diff
@@ -168,6 +168,141 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	return u, nil
 }
 
+// AddNewPaymentMethod - is a method to add new payment method for project
```
Comment (egonelbre) [inline satellite/console/service.go]: For all comments, the idiomatic Go way is to write it as a sentence where the method name is part of it. for example:
```

File: satellite/console/projectpayments.go [hunk 38]
```diff
@@ -13,16 +13,34 @@ import (
...
 	CreatedAt time.Time
 }
+
+// Card contains customer card info
+type Card struct {
+	Country  string
+	Brand    string
+	Name     string
+	ExpMonth int64
+	ExpYear  int64
```
Comment (egonelbre) [inline satellite/console/projectpayments.go]: ExpirationYear

File: satellite/console/projectpayments.go [hunk 39]
```diff
@@ -13,16 +13,34 @@ import (
...
 
 	CreatedAt time.Time
 }
+
+// Card contains customer card info
+type Card struct {
+	Country  string
+	Brand    string
+	Name     string
+	ExpMonth int64
```
Comment (egonelbre) [inline satellite/console/projectpayments.go]: ExpirationMonth

File: satellite/payments/stripepayments/stripe.go (hunk 32 above)
Comment (egonelbre) [reply satellite/payments/stripepayments/stripe.go]: Yeah, it looks like it's not using goimports. Usually you name the import explicitly if it doesn't match the path.

File: satellite/console/service.go (hunk 37 above)
Comment (Barterio) [reply satellite/console/service.go]: I made it to look consistent. Like `GenerateActivationToken`, `GeneratePasswordRecoveryToken`, `ActivateAccount` and others.

File: web/satellite/src/components/project/paymentMethods/NewPaymentMethodPopup.vue (hunk 24 above)
Comment (Barterio) [reply web/satellite/src/components/project/paymentMethods/NewPaymentMethodPopup.vue]: moved to environment variables

File: satellite/console/service.go (hunk 37 above)
Comment (Barterio) [reply satellite/console/service.go]: I can rename all occurrences at service, or just rename mine

File: satellite/console/projectpayments.go (hunk 4 above)
Comment (rikysya) [reply satellite/console/projectpayments.go]: No it's not

File: satellite/console/service.go [hunk 40]
```diff
@@ -168,6 +168,141 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+		Token:      paymentMethodToken,
+		CustomerID: string(info.CustomerID),
+	}
+
+	method, err := s.pm.AddPaymentMethod(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+
+	projectPaymentInfo := ProjectPayment{
```
Comment (rikysya) [inline satellite/console/service.go]: Why formatting is different here?

File: satellite/console/service.go [hunk 41]
```diff
@@ -168,6 +168,141 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+		ProjectID:       projectID,
+		PayerID:         authorization.User.ID,
+		PaymentMethodID: method.ID,
+		CreatedAt:       time.Now(),
+		IsDefault:       isDefault}
```
Comment (rikysya) [inline satellite/console/service.go]: Why formatting is different here?

File: satellite/console/service.go [hunk 42]
```diff
@@ -168,6 +168,142 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+		if err != nil {
+			return nil, err
+		}
+
+		cus, err := s.pm.GetCustomer(ctx, pm.CustomerID)
+		if err != nil {
+			return nil, err
+		}
+
+		user, err := s.store.Users().GetByEmail(ctx, cus.Email)
```
Comment (rikysya) [inline satellite/console/service.go]: that could fail

File: satellite/console/service.go [hunk 43]
```diff
@@ -168,6 +168,142 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+// SetDefaultPaymentMethod set default payment method for given project
+func (s *Service) SetDefaultPaymentMethod(ctx context.Context, projectPaymentID uuid.UUID, projectID uuid.UUID) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = GetAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	projectPayment, err := s.store.ProjectPayments().GetDefaultByProjectID(ctx, projectID)
```
Comment (rikysya) [inline satellite/console/service.go]: I think it should be one query like `db.UpdateDefault`, the rest is implementation detail.

File: satellite/console/service.go (hunk 42 above)
Comment (Barterio) [reply satellite/console/service.go]: this query returns only activated user

File: satellite/satellitedb/dbx/satellitedb.dbx (hunk 34 above)
Comment (Barterio) [reply satellite/satellitedb/dbx/satellitedb.dbx]: yes.
Since we have no data, the simplest migration is to drop and recreate new table version

File: satellite/console/service.go (hunk 42 above)
Comment (Barterio) [reply satellite/console/service.go]: Well, you were right. Found corner case when this is not a good approach

File: satellite/console/service.go [hunk 44]
```diff
@@ -168,6 +168,137 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+	}
+
+	var projectPayments []ProjectPayment
+	for _, payment := range projectPaymentInfos {
+		pm, err := s.pm.GetPaymentMethod(ctx, payment.PaymentMethodID)
+		if err != nil {
+			return nil, err
+		}
+
+		user, err := s.store.Users().Get(ctx, payment.PayerID)
```
Comment (rikysya) [inline satellite/console/service.go]: What for?

File: satellite/console/service.go [hunk 45]
```diff
@@ -168,6 +168,137 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
...
+	if err != nil {
+		return err
+	}
+	projectPayment.IsDefault = true
+
+	return s.store.ProjectPayments().Update(ctx, *projectPayment)
+}
+
+// DeletePaymentMethod deletes selected payment method
+func (s *Service) DeletePaymentMethod(ctx context.Context, projectPayment uuid.UUID) (err error) {
```
Comment (rikysya) [inline satellite/console/service.go]: DeleteProjectPaymentMethod

File: satellite/console/service.go (hunk 44 above)
Comment (Barterio) [reply satellite/console/service.go]: removed

File: satellite/console/service.go (hunk 45 above)
Comment (Barterio) [reply satellite/console/service.go]: renamed
